### PR TITLE
Refactor GC Object List Allocation/Initialization

### PR DESCRIPTION
**Pull request by RSalman**
Refactor initializeHeapRegionDescriptorExtension by separating allocation and initialization of each individual object list. This results in cleaner and reusable code. These changes are part of larger changes to reinit object lists for restore (CRIU).

Signed-off-by: Salman Rana <salman.rana@ibm.com>

### CODE REVIEW

**Comment by RSalman** [comment]
Please have a look @amicic @dmitripivkine

File: runtime/gc_glue_java/ConfigurationDelegate.hpp [hunk 1]
```diff
@@ -71,6 +72,66 @@ class MM_ConfigurationDelegate
  * Member functions
  */
 private:
+	MM_HeapRegionDescriptorStandardExtension *
+	allocateHeapRegionDescriptorExtension(MM_EnvironmentBase* env, uintptr_t listCount) {
```
Comment (amicic) [inline runtime/gc_glue_java/ConfigurationDelegate.hpp:76]: fix position of '*' (should be next to the variable, not type) throughout the code you've touched

**Comment by amicic** [comment]
looks ok, other then the minor formatting suggestion

File: runtime/gc_glue_java/ConfigurationDelegate.hpp (hunk 1 above)
Comment (RSalman) [reply runtime/gc_glue_java/ConfigurationDelegate.hpp]: Fixed my changes + other exiting code in the file

**Comment by amicic** [comment]
Jenkins test sanity xlinuxcriu jdk17

**Comment by amicic** [comment]
Jenkins compile win jdk8

**Comment by amicic** [comment]
Jenkins test sanity xlinuxcriu jdk17

**Pull request merged**